recipe/build-pyarrow.sh

```diff
@@ -38,6 +38,12 @@ if [[ "${target_platform}" == "linux-aarch64" ]]; then
     export PYARROW_CMAKE_OPTIONS="-DARROW_ARMV8_ARCH=armv8-a ${PYARROW_CMAKE_OPTIONS}"
 fi
 
+# Limit number of threads used to avoid hardware oversubscription
+if [[ "${target_platform}" == "linux-aarch64" ]] || [[ "${target_platform}" == "linux-ppc64le" ]]; then
+     export PYARROW_CMAKE_OPTIONS="${PYARROW_CMAKE_OPTIONS} -- -j 4"
```

Comment:
Not sure the right way to constrain threads used by the PyArrow build. Suggestions welcome 🙂

Reply:
Ah! Maybe we can use the [`CMAKE_BUILD_PARALLEL_LEVEL` environment variable]( https://cmake.org/cmake/help/latest/envvar/CMAKE_BUILD_PARALLEL_LEVEL.html )?